modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx

```diff
@@ -72,8 +73,9 @@ const FILTER_TYPES: Record<EFilterType, IFilterTypeProps> = {
 	[EFilterType.SELECTION]: {
 		Component: SelectionFilterFormContent,
 		availableFieldsFilter: (item: IField) =>
-			(item.type === EFieldType.STRING && !item.format) ||
-			item.type === EFieldType.INTEGER,
+			((item.type === EFieldType.STRING && !item.format) ||
+				item.type === EFieldType.INTEGER) &&
+			item.filterable,
```

Comment:
Do we expect any complex (non-leaf) field or arrays to be part of `xFilterable`? In that case, the restrictions on `item.type` shall change to allow them and exclude `(item.format === EFieldFormat.DATE ||item.format === EFieldFormat.DATE_TIME)`  so that date and selection filters offer disjoint subsets of fields for selection.